aviary/subsystems/propulsion/propeller_performance.py

```diff
@@ -214,22 +346,63 @@ def setup(self):
                 "density_ratio",
             ])
 
-        self.add_subsystem(
-            name='hamilton_standard',
-            subsys=HamiltonStandard(num_nodes=nn, aviary_options=aviary_options),
-            promotes_inputs=[
-                Dynamic.Mission.MACH,
-                "power_coefficient",
-                "advance_ratio",
-                "tip_mach",
-                Aircraft.Engine.PROPELLER_ACTIVITY_FACTOR,
-                Aircraft.Engine.PROPELLER_INTEGRATED_LIFT_COEFFICIENT,
-            ],
-            promotes_outputs=[
-                "thrust_coefficient",
-                "blade_angle",
-                "comp_tip_loss_factor",
-            ])
+        if use_propeller_map:
+            prop_model = PropellerMap('prop', aviary_options)
+            prop_file_path = aviary_options.get_val(
+                Aircraft.Engine.PROPELLER_DATA_FILE)
+            mach_type = prop_model.read_and_set_mach_type(prop_file_path)
+            if mach_type == OutMachType.HELICAL_MACH:
+                self.add_subsystem(
+                    name='selectedMach',
+                    subsys=OutMachs(
+                        num_nodes=nn, output_mach_type=OutMachType.HELICAL_MACH),
+                    promotes_inputs=[("mach", Dynamic.Mission.MACH), "tip_mach"],
+                    promotes_outputs=[("helical_mach", "selected_mach")],
+                )
+            else:
+                self.add_subsystem(
+                    name='selectedMach',
+                    subsys=om.ExecComp(
+                        'selected_mach = mach',
+                        mach={'units': 'unitless', 'shape': nn},
+                        selected_mach={'units': 'unitless', 'shape': nn},
+                        has_diag_partials=True,
+                    ),
+                    promotes_inputs=[("mach", Dynamic.Mission.MACH),],
+                    promotes_outputs=["selected_mach"],
+                )
```

Comment:
I think this will work:
```suggestion
            if mach_type == OutMachType.HELICAL_MACH:
                self.add_subsystem(
                    name='selectedMach',
                    subsys=OutMachs(
                        num_nodes=nn, output_mach_type=OutMachType.HELICAL_MACH),
                    promotes_inputs=[("mach", Dynamic.Mission.MACH), "tip_mach"],
                    promotes_outputs=[("helical_mach", "selected_mach")],
                )
            else:
                self.connect(Dynamic.Mission.MACH, "selected_mach")
```

Reply:
The `if mach_type == OutMachType.HELICAL_MACH:` section is unchanged, we are just replacing the passthrough component in the `else:` with a connection

Reply:
It didn't work. The error message is: "Attempted to connect from 'mach' to 'selected_mach', but 'mach' is an input. All connections must be from an output to an input."

Reply:
oh, ok. then you can leave it as is

Reply:
If I move `mach_type = helical_mach` out of comment block, `read_data_file()` function call will fail. But I accept your suggestion of using while look so that users can have multiple lines of comments and have `mach_type = helical_mach` at any line within the block.